pkg/template/runtask_functions.go

```diff
@@ -0,0 +1,212 @@
+// ---------
+// {{- get cstor volume | run -}}
+// {{- list cstor volume | run -}}
+func cstor() RunCommandMiddleware {
+	return CstorCategory()
+}
+
+// volume returns a new instance of volume based runtask command
+//
+// Examples:
+// ---------
+// {{- get cstor volume | run -}}
+// {{- list cstor volume | run -}}
+// {{- update jiva volume | run -}}
+// {{- create jiva volume | run -}}
+func volume() RunCommandMiddleware {
+	return VolumeCategory()
+}
+
+// slect returns the values of the specified paths post runtask command
+// execution
+//
+// Examples:
+// ---------
+// {{- select "all" | get cstor volume | run -}}
+// {{- select "name" | list cstor volume | run -}}
+// {{- select "name" "namespace" | update jiva volume | run -}}
+// {{- select ".metadata.name" | create jiva volume | run -}}
+func slect(paths ...string) RunCommandMiddleware {
+	if len(paths) == 0 {
+		paths = append(paths, "all")
+	}
+	return Select(paths)
+}
+
+// url sets the provided url as a input data to runtask command
+//
+// Examples:
+// ---------
+// {{- $url := "http://10.10.10.10:9501" -}}
+// {{- delete jiva volume | url $url | run -}}
+func url(u string, given *RunCommand) (updated *RunCommand) {
+	return WithData(given, "url", u)
+}
+
+// name sets the provided name as a input data to runtask command
+//
+// Examples:
+// ---------
+// {{- $url := "http://10.10.10.10:9501" -}}
+// {{- delete jiva volume | url $url | name "myvol" | run -}}
+func name(n string, given *RunCommand) (updated *RunCommand) {
+	return WithData(given, "name", n)
+}
+
+// run executes the runtask command
+//
+// Examples:
+// ---------
+// {{- get jiva volume | run -}}
+func run(given *RunCommand) RunCommandResult {
+	return given.Run()
+}
+
+// runlog executes the runtask command and stores the result as well as other
+// information at the provided paths
+//
+// Examples:
+// ---------
+// {{- get jiva volume | runlog "getj.result" "getj.extras" .Values -}}
+//
+// NOTES:
+// -----------
+// '.Values' is of type map[string]interface{} & is provided to the
+// template before template execution
+//
+// Once above template gets executed '.Values' will have .Values.getj.result
+// and .Values.getj.extras set with result and extras due to runtask command
+// execution
+func runlog(resultpath, debugpath string, store map[string]interface{}, given *RunCommand) (res RunCommandResult) {
+	res = run(given)
+
+	resultpath = strings.TrimPrefix(resultpath, ".")
+	debugpath = strings.TrimPrefix(debugpath, ".")
+
+	// result store
+	resultlocator := strings.Split(resultpath, ".")
+	util.SetNestedField(store, res.Result(), resultlocator...)
+
+	// debug store
+	debuglocator := strings.Split(debugpath, ".")
+	util.SetNestedField(store, res.Debug(), debuglocator...)
+	return
+}
+
+// runCommandFuncs returns the set of runtask command based template functions
+func runCommandFuncs() template.FuncMap {
+	return template.FuncMap{
+		"delete": delete,
+		"get":    get,
+		"lst":    list,
```

Comment:
`list ` will be easy to understand ?

Reply:
`list` is already a function in `sprig`. Trying to avoid overriding the behaviour.